app/js/controllers/main.js

```diff
@@ -50,22 +48,34 @@ export default class MainController extends Controller {
       return;
     }
 
-    this.view.customizer.setRootNode(document.documentElement);
+    // Insert the customizer view into the app's document
     document.body.appendChild(this.view.el);
 
-    var padding = this.view.el.querySelector('fxos-customizer').offsetHeight;
-    this._styleEl = document.createElement('style');
-    this._styleEl.textContent = `
-    html, body {
-      overflow: initial !important;
-    }
+    // And display it. This has to be in a setTimeout or the CSS animation
+    // won't work correctly since we just added it above.
+    setTimeout(() => { this.view.show(); });
+
+    // Wait a bit before populating it and defining the new stylesheet
+    // so that we can just get it on the screen quickly and don't interfere
+    // with the animation
+    setTimeout(() => {
+      this.view.customizer.setRootNode(document.documentElement);
+
+      var padding = this.view.el.querySelector('fxos-customizer').offsetHeight;
+      this._styleEl = document.createElement('style');
+      this._styleEl.textContent = `
+      html, body {
+        overflow: initial !important;
+      }
 
-    body {
-      padding-bottom: ${padding}px !important;
-    }`;
-    document.head.appendChild(this._styleEl);
+      body {
+        padding-bottom: ${padding}px !important;
+      }`;
+      document.head.appendChild(this._styleEl);
 
-    this.attached = true;
+      this.attached = true;
+      this.waitToBeClosed(); // We're open, so expect a close gesture
+    }, this.view.animationLength);
```

Comment:
Okay, I'll change this. I assume you mean transitionend, though, or does animationend work for both animations and transitions?

Another reason to use setTimeout() would be if we want to start populating the tree during the animation sometime.  I tried a 100ms timeout with a 200ms transition though and it seemed to interfere, so just waiting for the event is probably the right thing in general.